lib/travis/build/data.rb

```diff
@@ -143,12 +143,16 @@ def source_https?
 
       def source_ssh?
```

Comment:
Im wondering may you create one condition like
`!prefer_https? &&
((repo_private? || force_private?) && !installation?) ||
(repo_private? && custom_ssh_key?)`